### PR TITLE
Update to 16 Bit image compatability

### PR DESCRIPTION
**Pull request by honkmaster**

### CODE REVIEW

**Comment by honkmaster** [comment]
Found a bug ....

File: modules/olvis3/data/templates/filter/src/ThresholdFilter.cpp [hunk 1]
```diff
@@ -39,21 +41,35 @@ ThresholdFilter::ThresholdFilter()
...
 }
 
 void ThresholdFilter::execute()
 {
     int threshold = mThreshold;
-    const GrayImage src = mIn;
-    GrayImage dest;
-    cv::threshold(src, dest, threshold, 255, mMode);
+    cv::Mat src = mIn;
+    ((Image)src).convertToGray(CV_8U);
```
Comment (mariusknaust) [inline modules/olvis3/data/templates/filter/src/ThresholdFilter.cpp]: Except for Otsu’s algorithm cv::threshold should be able to handle CV_16SC1 (https://stackoverflow.com/questions/33099490/how-to-threshold-an-image-32-bit-using-otsu-s-binarization)

**Comment by honkmaster** [comment]
> @param src input array (single-channel, 8-bit or 32-bit floating point).

 Threshold does only handle 8 or 32, not 16. Thus, we need to convert up to 32 and afterwards convert down to the start bit width. Is this reasonable?

**Comment by honkmaster** [comment]
BTW: Why do they explicitly state only 8 and 32 bit ...

**Pull request closed without merging**